src/AutoRest.CSharp/Common/Output/Models/Types/SignatureType.cs

```diff
@@ -0,0 +1,318 @@
+            }
+
+            _typeFactory = typeFactory;
+            Methods = methods;
+            _defaultNamespace = defaultNamespace;
+            _defaultName = defaultName;
+            if (sourceInputModel is not null)
+            {
+                _customization = new SignatureType(typeFactory, PopulateMethodsFromCompilation(sourceInputModel?.Customization), null, defaultNamespace, defaultName);
+                _baselineContract = new SignatureType(typeFactory, PopulateMethodsFromCompilation(sourceInputModel?.PreviousContract), null, defaultNamespace, defaultName);
+                _methodChangeset ??= CompareMethods(Methods.Union(_customization?.Methods ?? Array.Empty<MethodSignature>(), MethodSignature.ParameterAndReturnTypeEqualityComparer), _baselineContract?.Methods);
+            }
+        }
+
+        private IReadOnlyList<Method>? _overloadMethods;
+        public IReadOnlyList<Method> OverloadMethods => _overloadMethods ??= EnsureOverloadMethods();
+
+        private IReadOnlyList<Method> EnsureOverloadMethods()
+        {
+            var overloadMethods = new List<Method>();
+            var updated = _methodChangeset?.Updated;
+            if (updated is null)
+            {
+                return Array.Empty<Method>();
+            }
+
+            foreach (var (current, previous) in updated)
+            {
+                if (TryGetPreviousMethodWithLessOptionalParameters(current, previous, out var currentMethodToCall, out var missingParameters))
+                {
+                    var overloadMethodSignature = new OverloadMethodSignature(currentMethodToCall, previous.WithParametersRequired(), missingParameters, previous.Description);
+                    overloadMethods.Add(new Method(overloadMethodSignature.PreviousMethodSignature with { IsHiddenFromUser = true }, BuildOverloadMethodBody(overloadMethodSignature)));
+                }
+            }
+            return overloadMethods;
+        }
+
+        private MethodBodyStatement BuildOverloadMethodBody(OverloadMethodSignature overloadMethodSignature)
+            => Return(new InvokeInstanceMethodExpression(null, overloadMethodSignature.MethodSignature.Name, BuildOverloadMethodParameters(overloadMethodSignature), null, overloadMethodSignature.PreviousMethodSignature.Modifiers.HasFlag(MethodSignatureModifiers.Async)));
+
+        private IReadOnlyList<ValueExpression> BuildOverloadMethodParameters(OverloadMethodSignature overloadMethodSignature)
+        {
+            var parameters = new List<ValueExpression>();
+            var set = overloadMethodSignature.MissingParameters.ToHashSet(Parameter.TypeAndNameEqualityComparer);
+            foreach (var parameter in overloadMethodSignature.MethodSignature.Parameters)
+            {
+                if (set.Contains(parameter))
+                {
+                    parameters.Add(new PositionalParameterReference(parameter.Name, Default.CastTo(parameter.Type)));
```

Comment:
We usually prefer null if the type could accept null instead of using default everywhere.
I think we could have a `GetDefaultExpression` method on `CSharpType` maybe - something like this:
```csharp
public partial class CSharpType
{
	public ValueExpression GetDefaultExpression()
	{
		if (IsValueType && !IsNullable)
			return Snippets.Default.CastTo(this);

		return Snippets.Null.CastTo(this);
	}
}
```
or introduce a method in `Snippets` so that we could have a null or default with type
We could list this as a follow up refactor because in the sample generation part we have a lot of if else to get default or null with types.

Reply:
Would like to postpone this refactoring later when we have this `GetDefaultExpression` in place.

Reply:
yeah we could do that later